storage/client_test.go

```diff
@@ -124,6 +126,41 @@ func TestGetSetTestIamPolicyEmulated(t *testing.T) {
 	})
 }
 
+func TestListBucketsEmulated(t *testing.T) {
+	transportClientTest(t, func(t *testing.T, project, bucket string, client storageClient) {
+		prefix := time.Now().Nanosecond()
+		want := []*BucketAttrs{
+			{Name: fmt.Sprintf("%d-%s-%d", prefix, bucket, time.Now().Nanosecond())},
+			{Name: fmt.Sprintf("%d-%s-%d", prefix, bucket, time.Now().Nanosecond())},
+		}
+		// Create the buckets that will be listed.
```

Comment:
Is prefix taken into account by the emulator? Can we create a bucket that should not be included in the listed objects?

For a really robust test, I'd think we'd need to have more buckets than the page limit as well to test calling fetch more than once. But this can also wait until we are ready to do integration tests.

Reply:
> Is prefix taken into account by the emulator? 

Yes it is (I had to add it though).

> Can we create a bucket that should not be included in the listed objects?

Yes I can add that! That's better than depending on the fact that other tests create buckets thus creating extra buckets to ignore with the prefix.

Reply:
Added an extra, non-prefixed bucket to the test data. Done.